scripts/create-subcounty-users.ts

```diff
@@ -0,0 +1,127 @@
+import { Logger } from '../src/utils/logger';
+import { generatePassword } from '../src/utils/password.utils';
+import { PermissionService } from '../src/service/permission-service';
+import { readUsersFromFile } from '../src/repository/csv-util';
+import { RoleService } from '../src/service/role-service';
+import { RLSService } from '../src/service/rls-service';
+import { UpdateRLSRequest } from '../src/types/rls';
+import { User } from '../src/types/user';
+import { UserService } from '../src/service/user-service';
+
+const CSV_FILENAME = './subcounty-users.csv';
+
+const BASE_SUBCOUNTY_ROLE_ID = 3585;
+
+type SubcountyAccountRequest = {
+  county: string;
+  subcounty?: string;
+  email: string;
+};
+
+type UserCreationResult = {
+  username: string;
+  password: string;
+  email: string;
+};
+
+async function getBasePermissions(): Promise<number[]> {
+  const permissionService = new PermissionService();
+  return await permissionService.getPermissionsByRoleId(BASE_SUBCOUNTY_ROLE_ID);
+}
+
+function subcountyUsername(request: SubcountyAccountRequest): string {
+  const transform = (val: string) => val.toLowerCase().replace(/ /g, '_').trim();
+  let result = `${transform(request.county)}`;
+  if (request.subcounty) {
+    result += `_${transform(request.subcounty)}`;
+  }
+  
+  return result;
+}
+
+async function createRole(subcountyRequest: SubcountyAccountRequest): Promise<number> {
+  Logger.info('Creating role from base...');
+  const basePermissionIds = await getBasePermissions();
+  const roleService = new RoleService();
+  const roleName = `subcounty_${subcountyUsername(subcountyRequest)}`;
+  const roles = await roleService.createRoles([roleName]);
+  await roleService.updateRolePermissions(roles, basePermissionIds);
+  return roles[0].id;
+}
+
+function getRLSClause(subcountyRequest: SubcountyAccountRequest): string {
+  let clause = `county='${subcountyRequest.county}'`;
+  if (subcountyRequest.subcounty) {
+    clause += ` AND subcounty_name='${subcountyRequest.subcounty}'`;
+  }
+
+  return clause;
+}
+
+async function createRLSPolicy(subcountyRequest: SubcountyAccountRequest, roleId: number) {
+  Logger.info('Creating an RLS...');
+  const rlsService = new RLSService();
+  const clause = getRLSClause(subcountyRequest);
+  const rlsRequest: UpdateRLSRequest = {
+    name: `subcounty_${subcountyUsername(subcountyRequest)}`,
+    group_key: '',
+    clause,
+    description: `RLS Policy for Subcounty ${subcountyRequest.county}.${subcountyRequest.subcounty}`,
+    filter_type: 'Regular',
+    roles: [roleId],
+    tables: [241],
```

Comment:
Are these users only intended to have access to the digital payment dashboards only or in addition to the [Supervisor dashboard](https://superset.echis.go.ke/superset/dashboard/535/)?

Reply:
i've never seen any request for them to have access to the supervisor dashboards. do you think that is desired?

Reply:
Yes, they should also have access to the Supervisor dashboard as well. I've confirmed that this is the case with Christine.

Reply:
OK! I'm doing a training on the Digital Payment dashboards tomorrow and I'll expand their permissions after that training.

Reply:
Great. Thanks. I'll approve this one and we can track that separately.